funnel/models/proposal.py

```diff
@@ -200,45 +209,45 @@ def waitlist(self):
         pass
 
     @with_roles(call={'admin'})
-    @state.transition(state.SUBMITTED, state.SHORTLISTED, title=__("Shortlist"), message=__("This proposal has been shortlisted"), type='success')
-    def shortlist(self):
-        pass
-
-    @with_roles(call={'admin'})
-    @state.transition(state.SUBMITTED, state.REJECTED, title=__("Reject"), message=__("This proposal has been rejected"), type='danger')
+    @state.transition(state.REJECTABLE, state.REJECTED, title=__("Reject"), message=__("This proposal has been rejected"), type='danger')
     def reject(self):
         pass
 
-    @with_roles(call={'admin'})
-    @state.transition(state.SUBMITTED, state.CANCELLED, title=__("Cancel"), message=__("This proposal has been cancelled"), type='danger')
+    @with_roles(call={'speaker', 'proposer'})
+    @state.transition(state.CANCELLABLE, state.CANCELLED, title=__("Cancel"), message=__("This proposal has been cancelled"), type='danger')
     def cancel(self):
         pass
 
-    @with_roles(call={'reviewer'})
+    @with_roles(call={'admin', 'reviewer'})
     @state.transition(state.SUBMITTED, state.AWAITING_DETAILS, title=__("Awaiting details"), message=__("Awaiting details for this proposal"), type='primary')
     def awaiting_details(self):
         pass
 
-    @with_roles(call={'admin'})
+    @with_roles(call={'admin', 'reviewer'})
     @state.transition(state.EVALUATEABLE, state.UNDER_EVALUATION, title=__("Under evaluation"), message=__("This proposal has been put under evaluation"), type='success')
     def under_evaluation(self):
         pass
 
-    @with_roles(call={'admin'})
-    @state.transition(state.SHORLISTABLE, state.SHORTLISTED_FOR_REHEARSAL, title=__("Shortlist for rehearsal"), message=__("This proposal has been shortlisted for rehearsal"), type='success')
-    def shortlist_for_rehearsal(self):
-        pass
-
-    @with_roles(call={'admin'})
-    @state.transition(state.SHORTLISTED_FOR_REHEARSAL, state.REHEARSAL, title=__("Rehearsal ongoing"), message=__("Rehearsal is now ongoing for this proposal"), type='success')
-    def rehearsal_ongoing(self):
-        pass
-
-    @with_roles(call={'admin', 'speaker', 'proposer'})
+    @with_roles(call={'speaker', 'proposer'})
     @state.transition(state.DELETABLE, state.DELETED, title=__("Delete"), message=__("This proposal has been deleted"), type='danger')
     def delete(self):
         pass
 
+    # @with_roles(call={'admin'})
```

Comment:
Leave the same note here about why these states have been removed. Ideally, issue a migration to move any proposals having this state into a more relevant state, and just remove all this code (separate PR).